### PR TITLE
Random.port and a few certificate helpers

### PR DESCRIPTION
**Pull request by ph**
This PR is a port of the tools I have created for the
logstash-input-beats to helper write better and easier tests.
- Allow to get a random available port from a range
- Allow to create chain of certificates

### CODE REVIEW

**Comment by guyboertje** [comment]
LGTM on the random port part.

Needs another review on the SSL certs part.

File: lib/flores/random.rb [hunk 1]
```diff
@@ -20,6 +20,46 @@
...
+  DEFAULT_PORT_CHECK_TIMEOUT = 1
+  DEFAULT_MAXIMUM_PORT_FIND_TRY = 15
+
+
+  # Try to find the first available port from a range.
+  # It will try 15 times and fails, this should give us some room
+  # if something wrong is going on.
+  def self.port(range = DEFAULT_PORT_RANGE)
+    try = 0
+    while try < DEFAULT_MAXIMUM_PORT_FIND_TRY
```
Comment (jordansissel) [inline lib/flores/random.rb]: I prefer `Fixnum#times` for this:

```ruby
DEFAULT_MAXIMUM_PORT_FIND_TRY.times do 
  candidate = integer(range)
  return candidate if port_available?(candidate)
end

raise ...
```

No refactor is required for this PR, though.

**Comment by jordansissel** [comment]
Can you add some specs to cover this?

**Comment by jordansissel** [comment]
BTW, I am OK merging this w/o tests in order to help get Logstash's tests stable, but would like specs for this as soon as we can.

**Pull request closed without merging**